ProteoformSuiteInternal/TdBuReader.cs

```diff
@@ -0,0 +1,222 @@
+                    foreach (Row row in rowcollection)
+                    {
+                        if (row.RowIndex.Value == 1) continue;
+                        IEnumerable<Cell> cells = GetRowCells(row);
+                        List<string> cellStrings = new List<string>();
+                        foreach (Cell cell in cells)
+                        {
+                            cellStrings.Add(GetCellValue(spreadsheetDocument, cell));
+                        }
+                        //get ptms on proteoform
+                        List<Ptm> ptm_list = new List<Ptm>();
+                        //N-term modifications
+                        if (cellStrings[10].Length > 0)
+                        {
+                            int position = 0;
+                            if (cellStrings[10].Split(':')[1] == "1458")
+                            {
+                                ModificationWithMass mod = Lollipop.uniprotModificationTable.Values.SelectMany(m => m).OfType<ModificationWithMass>().Where(m => m.id.Contains("acetyl") && m.motif.Motif == cellStrings[4][0].ToString()).FirstOrDefault();
+                                if (mod != null)
+                                {
+                                    ptm_list.Add(new Ptm(position, mod));
+                                }
+                                //found one case where PTM not in ptmlist.txt (acetylasparagine)
+                                else
+                                {
+                                    ptm_list.Add(new Ptm(position, new ModificationWithMass("N-acetylation", null, null, ModificationSites.NTerminus, 0, null, -1, null, null, null)));
+                                }
+                            }
+                        }
+                        //don't have example of c-term modification to write code
+                        //other mods
+                        if (cellStrings[9].Length > 0)
+                        {
+                            string[] res_ids = cellStrings[9].Split('|');
+                            foreach (string new_ptm in res_ids)
+                            {
+                                string resid = new_ptm.Split(':')[1].Split('@')[0];
+                                while (resid.Length < 4) resid = "0" + resid;
+                                resid = "AA" + resid;
+                                int position = Convert.ToInt16(new_ptm.Split(':')[1].Split('@')[1]);
+                                ModificationWithMass mod = Lollipop.uniprotModificationTable.Values.SelectMany(m => m).OfType<ModificationWithMass>().Where(m => m.linksToOtherDbs.ContainsKey("RESID")).Where(m => m.linksToOtherDbs["RESID"].Contains(resid)).FirstOrDefault();
+                                if (mod != null) ptm_list.Add(new Ptm(position, mod));
+                            }
+                        }
+
+                        Result_Set result_set = new Result_Set();
```

Comment:
What is this `Result_Set` class? If this is one of yours, please consider choosing a more descriptive name. 😄